### PR TITLE
Feature/3010 static content sitemap

### PR DESCRIPTION
**Pull request by philipkcl**
# Ensure existing static content is reflected in the sitemap

*Please don't delete any sections when completing this PR template; instead enter **N/A** for checkboxes or sections which are not applicable, unless otherwise stated below*

See # https://github.com/DOAJ/doajPM/issues/3010

Describe the scope/purpose of the PR here in as much detail as you like

* all static url reference endpoint of cms/data/nav.xml 
* if there are some endpoint is invalid, will be marked in log and ignore it .

## Categorisation

This PR...
- [ ] has scripts to run
- [ ] has migrations to run
- [ ] adds new infrastructure
- [ ] changes the CI pipeline
- [ ] affects the public site
- [ ] affects the editorial area
- [ ] affects the publisher area

## Basic PR Checklist

- [ ] FeatureMap annotations have been added
- [x] Unit tests have been added/modified
- [ ] Functional tests have been added/modified
- [x] Code has been run manually in development, and functional tests followed locally
- [x] No deprecated methods are used
- [x] No magic strings/numbers - all strings are in `constants` or `messages` files
- [ ] ES queries are wrapped in a Query object rather than inlined in the code
- [ ] Where possible our common library functions have been used (e.g. dates manipulated via `dates`)
- [ ] If needed, migration has been created and tested locally
- [ ] Release sheet has been created, and completed as far as is possible https://docs.google.com/spreadsheets/d/1Bqx23J1MwXzjrmAygbqlU3YHxN1Wf7zkkRv14eTVLZQ/edit
- [ ] Documentation updates - if needed - have been identified and prepared for inclusion into main documentation (e.g. added and highlighted/commented as appropriate to this PR)
    - [ ] Core model documentation: https://docs.google.com/spreadsheets/d/1lun2S9vwGbyfy3WjIjgXBm05D-3wWDZ4bp8xiIYfImM/edit
    - [ ] Events and consumers documentation: https://docs.google.com/spreadsheets/d/1oIeG5vg-blm2MZCE-7YhwulUlSz6TOUeY8jAftdP9JE/edit
- [ ] There has been a recent merge up from `develop` (or other base branch)

## Testing

List the Functional Tests that must be run to confirm this feature

1. ...
2. ...

## Deployment

What deployment considerations are there? (delete any sections you don't need)

### Configuration changes

* remove variable `STATIC_PAGES` from settings.py

### CODE REVIEW

File: portality/bll/services/site.py [hunk 1]
```diff
@@ -1,13 +1,57 @@
...
 from portality.bll import exceptions
+from portality.core import app
+from portality.lib.argvalidate import argvalidate
+from portality.store import StoreFactory, prune_container
+from portality.util import get_full_url_by_endpoint
+
+NS = "{http://www.sitemaps.org/schemas/sitemap/0.9}"
+
+
+def create_simple_sub_element(parent, element_name, text=None):
```
Comment (richard-jones) [inline portality/bll/services/site.py:18]: Propose to move this function to `portality.xmlutil`

File: portality/bll/services/site.py [hunk 2]
```diff
@@ -1,13 +1,57 @@
...
+
+def yield_all_route(entries: Iterable[dict]):
+    for e in entries:
+        if 'route' in e:
+            yield e['route']
+        if 'entries' in e:
+            yield from yield_all_route(e['entries'])
+
+
+def get_url_safe(endpoint):
```
Comment (richard-jones) [inline portality/bll/services/site.py:44]: Propose to move this function to `portality.util` alongside our version of `url_for`

File: portality/bll/services/site.py [hunk 3]
```diff
@@ -1,13 +1,57 @@
...
+
+    create_simple_sub_element(url_ele, 'loc', loc)
+    if lastmod is not None:
+        create_simple_sub_element(url_ele, "lastmod", lastmod)
+    create_simple_sub_element(url_ele, "changefreq", change_freq)
+
+    return url_ele
+
+
+def yield_all_route(entries: Iterable[dict]):
```
Comment (richard-jones) [inline portality/bll/services/site.py:36]: Propose a new module like `portality.lib.nav` where this function could sit

File: portality/bll/services/site.py [hunk 4]
```diff
@@ -1,13 +1,57 @@
...
+
+def get_url_safe(endpoint):
+    try:
+        return get_full_url_by_endpoint(endpoint)
+    except werkzeug.routing.BuildError:
+        app.logger.warning(f'endpoint not found -- [{endpoint}]')
+        return None
+
+
+def get_nav_entries() -> Iterable[dict]:
```
Comment (richard-jones) [inline portality/bll/services/site.py:52]: Propose that this also go in `portality.lib.nav`

**Review by richard-jones — CHANGES_REQUESTED** [review]
This is excellent, works well.  Only changes I suggest are proposed reorganising of where some functions sit in the codebase.  I have annotated each function with where I think it could sit naturally alongside similar library functions.

File: portality/bll/services/site.py (hunk 1 above)
Comment (philipkcl) [reply portality/bll/services/site.py:18]: unfortunately, this function contain logic for sitemap.

let's extract or refactor in future if other module need function similar to `create_simple_sub_element`

**Comment by philipkcl** [comment]
updated for review

**Review by richard-jones — APPROVED** [review]

**Review by Steven-Eardley — APPROVED** [review]
Looks good & working well on test

**Pull request closed without merging**